### PR TITLE
Mate: bump to version 1.27

### PR DESCRIPTION
**Pull request by immolo**
Push mate to version 1.27 to match upstream.

### CODE REVIEW

**Comment by immolo** [comment]
Marking this as a draft for extra testing however currently this works very well on my stable amd64 laptop running gcc/glibc.

**Comment by immolo** [comment]
I have left out the meson changes for this version as getting a stable base to work from felt more important then worrying about about the 2 packages that support it. I have also left out mate-base/mate-wayland-session for now as the release notes make it seem too experimental to enable until further testing is complete.

My current agenda for MATE is as follows:

1.  1.27 merged as is for testing with users.
2. Check for ModernC comparability and upstream where needed.
3. Enable full Wayland Support
4. Convert to meson during version bumps as things stablise.

**Comment by immolo** [comment]
Failed checks are new packages added by this commit so will resolve themselves once merged.

**Comment by thesamesam** [comment]
> Failed checks are new packages added by this commit so will resolve themselves once merged.

I don't think that adds up. The CI checks the PR as a whole. I don't see mate-icon-theme in this PR?

File: mate-base/caja/caja-1.26.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Copyright 1999-2023 Gentoo Authors
```
Comment (thesamesam) [inline mate-base/caja/caja-1.26.3.ebuild:1]: In the `pkgdiff-mg -b ....` output for caja, I see:
```
[...]
+m4_define(gtk_layer_shell_minver,      0.8.0)
[...]
```

But I don't see a gtk-layer-shell dependency in the ebuild.

There is also a new Wayland configure option:
```
+AC_ARG_ENABLE(wayland,
+              [AS_HELP_STRING([--enable-wayland],
+                              [Explicitly enable or disable Wayland support
+                              (default is to enable only if Wayland client development library is detected)])],
+              [enable_wayland=$enableval],
+              [enable_wayland=auto])
+
+# Check if we have gtk-layer-shell installed, and thus should build with Wayland support
+have_wayland=no
+m4_define([WAYLAND_DEPS], [gtk-layer-shell-0 >= gtk_layer_shell_minver wayland-client])
+AS_IF([test "x$enable_wayland" = "xyes"],
+      [PKG_CHECK_MODULES([WAYLAND], [WAYLAND_DEPS], [have_wayland=yes])],
+      [test "x$enable_wayland" != "xno"],
+      [PKG_CHECK_MODULES([WAYLAND], [WAYLAND_DEPS], [have_wayland=yes], [have_wayland=no])])
+
+AM_CONDITIONAL(ENABLE_WAYLAND, [test "x$have_wayland" = "xyes"])
+
+if test "x$have_wayland" = "xyes"; then
+  AC_DEFINE(HAVE_WAYLAND, 1, [Have the Wayland development library])
+fi
```
... that's not wired up either yet.

Comment (thesamesam) [reply mate-base/caja/caja-1.26.3.ebuild:1]: Please make sure there's not other stuff like this missing from other pkgs.

File: app-arch/engrampa/engrampa-1.27.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+DEPEND="
+	>=dev-libs/glib-2.50:2
+	>=dev-libs/json-glib-0.14
+	virtual/libintl
+	x11-libs/gdk-pixbuf:2
+	>=x11-libs/gtk+-3.22:3[X]
+	x11-libs/libSM
+	x11-libs/pango
+	caja? ( >=mate-base/caja-1.17.1 )
+	magic? ( sys-apps/file )
```
Comment (thesamesam) [inline app-arch/engrampa/engrampa-1.27.1.ebuild:32]: From the diff:
```
-       dnl *******************************************
-       dnl This check can be removed with file >= 5.38
-       dnl *******************************************
-       AC_MSG_CHECKING([for zstd mime type])
-       AC_RUN_IFELSE([AC_LANG_PROGRAM([[#include <stdio.h>
-#include <string.h>
-#include <magic.h>]],
-               [[
-    FILE *fptr;
- ````

so I think we want to depend on >=5.38 for file now that they removed the check.

File: mate-base/caja/caja-1.26.3.ebuild (hunk 1 above)
Comment (immolo) [reply mate-base/caja/caja-1.26.3.ebuild:1]: I left it out of 1.26.x builds but is enabled in all 1.27 which I found.

Comment (thesamesam) [reply mate-base/caja/caja-1.26.3.ebuild:1]: Oh, I didn't spot you were doing 1.26.x as well, ok

Comment (immolo) [reply mate-base/caja/caja-1.26.3.ebuild:1]: I did a few with important bug fixes which may help, I can drop these and just push 1.27 if that makes more sense.

File: mate-base/mate-applets/mate-applets-1.27.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,98 @@
+# Copyright 1999-2023 Gentoo Authors
```
Comment (thesamesam) [inline mate-base/mate-applets/mate-applets-1.27.1.ebuild:1]: Looks like this bump will fix https://bugs.gentoo.org/919089, so tag that (mentioned in release notes)

File: mate-base/mate-control-center/mate-control-center-1.27.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+MATE_LA_PUNT="yes"
+
+inherit mate
+
+if [[ ${PV} != 9999 ]]; then
+	KEYWORDS="amd64 ~arm ~arm64 ~loong ~riscv x86"
```
Comment (thesamesam) [inline mate-base/mate-control-center/mate-control-center-1.27.0.ebuild:11]: ekeyword ~all, don't commit as stable

File: mate-base/caja/caja-1.26.3.ebuild (hunk 1 above)
Comment (thesamesam) [reply mate-base/caja/caja-1.26.3.ebuild:1]: nah, it's ok, I just didn't cotton on

File: mate-base/mate-menus/mate-menus-1.27.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+GNOME2_LA_PUNT="yes"
+
+inherit mate
+
+if [[ ${PV} != 9999 ]]; then
+	KEYWORDS="amd64 ~arm ~arm64 ~loong ~riscv x86"
```
Comment (thesamesam) [inline mate-base/mate-menus/mate-menus-1.27.0.ebuild:11]: not stable

File: mate-extra/caja-actions/caja-actions-1.27.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+fi
+
+DESCRIPTION="Caja Actions"
+HOMEPAGE="https://github.com/mate-desktop/caja-actions"
+
+LICENSE="GPL-2+"
+SLOT="0"
+
+COMMON_DEPEND="
+	>=dev-libs/glib-2.32:2
```
Comment (thesamesam) [inline mate-extra/caja-actions/caja-actions-1.27.0.ebuild:21]:
```
-glib_required=2.32.1
+glib_required=2.66.0
```

File: mate-extra/caja-dropbox/caja-dropbox-1.27.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,88 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{9,10,11} )
```
Comment (thesamesam) [inline mate-extra/caja-dropbox/caja-dropbox-1.27.0.ebuild:6]: drop 9, it's a noop now

File: mate-extra/caja-extensions/caja-extensions-1.27.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,74 @@
+# Copyright 1999-2023 Gentoo Authors
```
Comment (thesamesam) [inline mate-extra/caja-extensions/caja-extensions-1.27.0.ebuild:1]: need to wire up the new av extension?

File: mate-extra/mate-polkit/mate-polkit-1.27.1.ebuild [hunk 9]
```diff
@@ -0,0 +1,41 @@
+# Copyright 1999-2023 Gentoo Authors
```
Comment (thesamesam) [inline mate-extra/mate-polkit/mate-polkit-1.27.1.ebuild:1]: This isn't something I would've expected you to spot yet, as it's a bit niche, but:
```
+if  test "x$enable_appindicator" == "xauto" &&
+   (test "x$have_appindicator_ayatana" == "xyes" ||
+    test "x$have_appindicator_ubuntu" == "xyes"); then
```

These should be `=`, not `==`, or it's going to break with dash as /bin/sh. We should send that upstream.

**Comment by immolo** [comment]
All fixes are now included.

Before merging also be aware 1.27 is the development release cycle so I'll try to keep 1.26 bug fixes applied until 1.28 is out and keep 1.27 as our testing version. This means users can choose between an upstream approved version to run or a version which will help find issues early while making it easier for llvm systems to run MATE. This change will be made clear in a wiki draft I have prepared one this merged.

For everyone else patches and bug reports are very much welcome.

**Comment by Kangie** [comment]
:heart:

**Comment by thesamesam** [comment]
My only further comment is that we usually don't put development releases in ~arch. They should normally be masked or unkeyworded. We _do_ put them in ~arch for XFCE (although I don't like that) and it's led to problems as well.

It leads to poor UX if users hit a problem, but it's also a problem if you're not interested in MATE maintenance for the long-haul, because suppose you get bored of MATE, it means we've now got development releases sitting in ~arch forever, as opposed to some nice stable releases which are supposed to be fine.

I'm not hard against it though if you can make a case for it.

**Comment by immolo** [comment]
> My only further comment is that we usually don't put development releases in ~arch. They should normally be masked or unkeyworded. We _do_ put them in ~arch for XFCE (although I don't like that) and it's led to problems as well.

I've been wrestling with the same thought process myself after learning how the releases work upstream.

The reason I went for ~arch in the end was the fact it's going to be the version I will run and will be doing all the fixes in that version. I'm going to be heavily relying on users to report issues in 1.26 although will keep an eye on them to push bug fixes while I'm there. I fully agree with what you said though about this being an issue if I stop working on MATE as while the DE wasn't in the best shape at least it worked for a year.  

I feel that as we both have had the same worry the best way forward would be to unkeyword for now then see how community helps with reporting bugs to keep 1.26 stable and reopen this question at a later date if one of us feels it's not working.

**Comment by Kangie** [comment]
> I've been wrestling with the same thought process myself after learning how the releases work upstream.
>  ...
> I feel that as we both have had the same worry the best way forward would be to unkeyword for now then see how community helps with reporting bugs to keep 1.26 stable and reopen this question at a later date if one of us feels it's not working.

Consider putting keyworded 1.27 ebuilds in a Mate repository though, get you some canary users and give you a good place to stage changes.

**Pull request closed without merging**